src/allmydata/test/test_checker.py

```diff
@@ -164,11 +167,11 @@ def create_fake_client(self):
         return c
 
     def render_json(self, resource):
-        return resource.render(TestRequest(args={"output": ["json"]}))
+        return self.successResultOf(render(resource, {"output": ["json"]}))
```

Comment:
It's true `IResource.render` does exist but it is only part of the rendering process.  While it used to be a sufficient part, it no longer is because our resources *always* return `NOT_DONE_YET` now and deal with writing to the request directly (thanks to `@render_exception`).

`common_web.render` works in all cases where `resource.render` worked but it also has the extra logic to handle `NOT_DONE_YET`.  It is currently favorably inclined towards to Nevow but it is a chokepoint  which we can switch over to be more appropriate to Twisted at some point.

Reply:
If the application render function for the resource returns some JSON, that still gets turned into a NOT_DONE_YET result by the decorator.  Then the `render` helper used in these tests turns the NOT_DONE_YET result into a Deferred that fires with the JSON.